src/D2L.Security.OAuth2/Keys/PrivateKeyData.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+
+namespace D2L.Security.OAuth2.Keys {
+	public sealed class PrivateKeyData {
+		internal static class KeyKinds {
+			public const string Rsa = "rsa";
+			public const string Ecdsa = "ecdsa";
+		}
+
+		public PrivateKeyData(
+			string id,
+			string kind,
+			byte[] data,
+			DateTimeOffset createdAt,
+			DateTimeOffset notBefore,
+			DateTimeOffset expiresAt
+		) {
+			Id = id;
+			Kind = kind;
+			Data = data;
+			CreatedAt = createdAt;
+			NotBefore = notBefore;
+			ExpiresAt = expiresAt;
+		}
+
+		/// <summary>
+		/// The jwk "key id"/kid.
+		/// </summary>
+		public string Id { get; }
+
+		/// <summary>
+		/// The kind of key (e.g. RSA)
+		/// </summary>
+		public string Kind { get; }
+
+		/// <summary>
+		/// Kind-specific key data (including the private bits.)
+		/// </summary>
+		public byte[] Data { get; }
+
+		/// <summary>
+		/// The time the key was originally created.
+		/// </summary>
+		public DateTimeOffset CreatedAt { get; }
+
+		/// <summary>
+		/// The key should not be used to sign things before this point. The
+		/// gap between CreatedAt and NotBefore allows for jwks caches to
+		/// expire and gives opportunity for people to fetch our new key in the
+		/// background.
+		//
+		/// If there are no other keys available in an emergency it is okay
+		/// to sign messages with a key that has not reached NotBefore.
+		/// </summary>
+		public DateTimeOffset NotBefore { get; }

```

Comment:
I got rid of `BestBefore` and jitter + debouncing, for simplicity.

I think there is still value in that, but communicating all that information back to the client (e.g. the auth service) is a little tricky/would require some more API design. I don't think, given the things that will use this library and how they're implemented, we don't need to worry about thundering herd problems anymore (given what things we plan to shift to background jobs (i.e. shrinking the herd) + the number of processes we're talking about + the database where we are storing things).